examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -333,27 +383,134 @@ int main_(std::vector<std::string> const &args)
+}
+
+template <class NO>
+int run(std::vector<std::string> const &args, TimeMonitor &time_monitor,
+        MPI_Comm comm)
+{
+  Arguments arguments;
+
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+    desc.add_options()
+        ( "help", "produce help message" )
+        ( "values", bpo::value<int>(&arguments.n_values)->default_value(50000), "Number of indexable values (source) per MPI rank." )
+        ( "queries", bpo::value<int>(&arguments.n_queries)->default_value(20000), "Number of queries (target) per MPI rank." )
+        ( "neighbors", bpo::value<int>(&arguments.n_neighbors)->default_value(10), "Desired number of results per query." )
+        ( "shift", bpo::value<double>(&arguments.shift)->default_value(1.), "Shift of the point clouds. '0' means the clouds are built "
+                                                                  "at the same place, while '1' places the clouds next to each"
+                                                                  "other. Negative values and values larger than one "
+                                                                  "mean that the clouds are separated." )
+        ( "partition_dim", bpo::value<int>(&arguments.partition_dim)->default_value(3), "Number of dimension used by the partitioning of the global "
+                                                                              "point cloud. 1 -> local clouds are aligned on a line, 2 -> "
+                                                                              "local clouds form a board, 3 -> local clouds form a box." )
+        ( "do-not-perform-knn-search", "skip kNN search" )
+        ( "do-not-perform-radius-search", "skip radius search" )
+        ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(args).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  int comm_rank;
+  MPI_Comm_rank(comm, &comm_rank);
+  int comm_size;
+  MPI_Comm_size(comm, &comm_size);
 
-  return 0;
+  if (vm.count("help"))
+  {
+    if (comm_rank == 0)
+      std::cout << desc << '\n';
+    throw HelpPrinted();
+  }
+
+  if (vm.count("do-not-perform-knn-search"))
+    arguments.perform_knn_search = false;
+  if (vm.count("do-not-perform-radius-search"))
+    arguments.perform_radius_search = false;
+
+  if (comm_rank == 0)
+  {
+    std::cout << std::boolalpha;
+    std::cout << "\nRunning with arguments:\n"
+              << "perform knn search      : " << arguments.perform_knn_search
+              << '\n'
+              << "perform radius search   : " << arguments.perform_radius_search
+              << '\n'
+              << "#points/MPI process     : " << arguments.n_values << '\n'
+              << "#queries/MPI process    : " << arguments.n_queries << '\n'
+              << "size of shift           : " << arguments.shift << '\n'
+              << "dimension               : " << arguments.partition_dim << '\n'
+              << '\n';
```

Comment:
Please consider adding a "print" member function to `Arguments` or a "put to" operator that write that info into a stream.